src/plugins/visualizations/public/utils/saved_visualize_utils.ts

```diff
@@ -0,0 +1,362 @@
+  const { id, title, description, visState, uiStateJSON, searchSourceFields } = savedVis;
+
+  const aggs = searchSourceFields && searchSourceFields.index ? visState.aggs || [] : visState.aggs;
+
+  return {
+    id,
+    title,
+    type: visState.type,
+    description,
+    params: visState.params,
+    uiState: JSON.parse(uiStateJSON || '{}'),
+    data: {
+      aggs,
+      searchSource: searchSourceFields!,
+      savedSearchId: savedVis.savedSearchId,
+    },
+  };
+};
+
+export const convertFromSerializedVis = (vis: SerializedVis): ISavedVis => {
+  return {
+    id: vis.id,
+    title: vis.title,
+    description: vis.description,
+    visState: {
+      title: vis.title,
+      type: vis.type,
+      aggs: vis.data.aggs,
+      params: vis.params,
+    },
+    uiStateJSON: JSON.stringify(vis.uiState),
+    searchSourceFields: vis.data.searchSource,
+    savedSearchId: vis.data.savedSearchId,
+  };
+};
+
+export async function findListItems(
+  savedObjectsClient: SavedObjectsClientContract,
+  search: string,
+  size: number,
+  references?: SavedObjectsFindOptionsReference[]
+) {
+  const visTypes = getTypes();
+  const visAliases = visTypes.getAliases();
+  const extensions = visAliases
+    .map((v) => v.appExtensions?.visualizations)
+    .filter(Boolean) as VisualizationsAppExtension[];
+  const extensionByType = extensions.reduce((acc, m) => {
+    return m!.docTypes.reduce((_acc, type) => {
+      acc[type] = m;
+      return acc;
+    }, acc);
+  }, {} as { [visType: string]: VisualizationsAppExtension });
+  const searchOption = (field: string, ...defaults: string[]) =>
+    _(extensions).map(field).concat(defaults).compact().flatten().uniq().value() as string[];
+  const searchOptions: SavedObjectsFindOptions = {
+    type: searchOption('docTypes', 'visualization'),
+    searchFields: searchOption('searchFields', 'title^3', 'description'),
+    search: search ? `${search}*` : undefined,
+    perPage: size,
+    page: 1,
+    defaultSearchOperator: 'AND' as 'AND',
+    hasReference: references,
+  };
+
+  const { total, savedObjects } = await savedObjectsClient.find<SavedObjectAttributes>(
+    searchOptions
+  );
+
+  return {
+    total,
+    hits: savedObjects.map((savedObject) => {
+      const config = extensionByType[savedObject.type];
+
+      if (config) {
+        return {
+          ...config.toListItem(savedObject),
+          references: savedObject.references,
+        };
+      } else {
+        return mapHitSource(visTypes, savedObject);
+      }
+    }),
+  };
+}
+
+export async function getSavedVisualization(
+  services: {
+    savedObjectsClient: SavedObjectsClientContract;
+    search: any;
+    dataViews: any;
+  },
+  opts?: any
+) {
+  if (typeof opts !== 'object') {
+    opts = { id: opts };
+  }
+
+  const id = (opts.id as string) || '';
+  const savedObject = {
+    id,
+    displayName: SAVED_VIS_TYPE,
+    getEsType: () => SAVED_VIS_TYPE,
```

Comment:
This is probably true for a bunch more places, but I think some of that cruft is not necessary anymore.